common/persistence/persistenceInterface.go

```diff
@@ -112,6 +110,23 @@ type (
 		GetHistoryTree(request *GetHistoryTreeRequest) (*GetHistoryTreeResponse, error)
 	}
 
+	// VisibilityStore is the store interface for visibility
+	VisibilityStore interface {
+		Closeable
+		GetName() string
+		RecordWorkflowExecutionStarted(request *RecordWorkflowExecutionStartedRequest) error
```

Comment:
Lets follow the same pattern as History for request object to take in DataBlob rather than []byte and encoding.

Reply:
agree with do in https://github.com/uber/cadence/issues/1718